### PR TITLE
[FIX]: add check if there are reads in lane before adding fastq files to housekeeper

### PR DESCRIPTION
**Pull request by karlnyr**
## Description

Addresses https://github.com/Clinical-Genomics/cg/issues/3798

### Added

- Additional condition in post processing flow for Illimina that checks if the sample has any reads in the lane

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by karlnyr** [comment]
Should have a test for this

File: cg/services/illumina/post_processing/utils.py [hunk 1]
```diff
@@ -14,6 +14,11 @@
 
 LOG = logging.getLogger(__name__)
 
+def sample_is_negative_control_with_reads_in_lane(
+    is_negative_control: bool, metric: IlluminaSampleSequencingMetricsDTO
+) -> bool:
+    """Check if a sample is a negative control and has reads."""
```
Comment (diitaz93) [inline cg/services/illumina/post_processing/utils.py:20]:
```suggestion
def sample_is_negative_control_with_reads_in_lane(
    is_negative_control: bool, metric: IlluminaSampleSequencingMetricsDTO
) -> bool:
    """Check if a sample is a negative control and has reads on a lane."""
```

File: tests/services/illumina/post_processing/test_utils.py [hunk 2]
```diff
@@ -24,9 +24,43 @@
     get_q30_threshold,
     get_sample_fastqs_from_flow_cell,
     get_undetermined_fastqs,
+    sample_is_negative_control_with_reads_in_lane,
 )
 
 
+@pytest.mark.parametrize(
+    "reads, is_negative_control, is_negative_control_with_reads",
```
Comment (diitaz93) [inline tests/services/illumina/post_processing/test_utils.py:32]:
```suggestion
    "reads, is_negative_control, expected_negative_control_with_reads",
```

File: tests/services/illumina/post_processing/test_utils.py [hunk 3]
```diff
@@ -24,9 +24,43 @@
...
+    # GIVEN a metric with a specific amount of reads for a sample
+    mapped_metric.total_reads_in_lane = reads
+
+    # WHEN checking if the sample is a negative control with reads in the lane
+    result: bool = sample_is_negative_control_with_reads_in_lane(
+        is_negative_control=is_negative_control, metric=mapped_metric
+    )
+
+    # THEN the result should be return according to the expected outcome
+    assert result == is_negative_control_with_reads
```
Comment (diitaz93) [inline tests/services/illumina/post_processing/test_utils.py:61]:
```suggestion
    assert result == expected_negative_control_with_reads
```

File: cg/services/illumina/post_processing/housekeeper_storage.py [hunk 4]
```diff
@@ -48,6 +46,12 @@ def _should_fastq_path_be_stored_in_housekeeper(
     )
 
     if metric:
+
+        if sample_is_negative_control_with_reads_in_lane(
```
Comment (islean) [inline cg/services/illumina/post_processing/housekeeper_storage.py:50]:
```suggestion
        if is_sample_negative_control_with_reads_in_lane(
```
Nice convention to highlight that it returns a boolean

File: cg/services/illumina/post_processing/utils.py (hunk 1 above)
Comment (islean) [reply cg/services/illumina/post_processing/utils.py:21]: I think the docstring just adds confusion in this case 🙈

File: cg/services/illumina/post_processing/housekeeper_storage.py (hunk 4 above)
Comment (karlnyr) [reply cg/services/illumina/post_processing/housekeeper_storage.py:50]: agree

**Review by islean — APPROVED** [review]
💪

**Pull request merged**